experiments/clang_interface/README.md

```diff
@@ -0,0 +1,12 @@
+Clang interface experiment
+==========================
+
+This experiment is a quick-and-dirty hack to explore whether we can use clang to:
+
+ - Parse a C/C++ header.
+ - Cache the parse result so that the Verona compiler can do incremental builds.
+ - Resolve types and symbols.
+ - Instantiate templates.
+ - Synthesize functions that provide simple-ABI wrappers.
+
+It is incomplete and ugly code but serves to demonstrate that the clang APIs are sufficient for our purposes.
```

Comment:
Could you add a test or example of how to use this?  The value for others will be more significant if they can see it used briefly.

Reply:
This is not intended to be used as-is, it's to demonstrate how various bits of clang the APIs can be used.  As the README says, it is very incomplete and just contains example code for the things in the list.  If you run it, it will spit out some LLVM IR for a function in a compilation unit that includes some C++ things.

Reply:
If you add a 
>  If you run it, it will spit out some LLVM IR for a function in a compilation unit that includes some C++ things.

to the readme, possibly with a command line, then I would know what to expect.

Reply:
I honestly don't think there is any benefit to anyone in running this code.  The benefit is in reading it and incorporating bits of it elsewhere.  The only reason that I run it is to check that it works (for what it does).